packages/dev/core/src/AudioV2/webAudio/webAudioEngine.ts

```diff
@@ -0,0 +1,278 @@
+import { Observable } from "../../Misc/observable";
+import type { Nullable } from "../../types";
+import type { AbstractNamedAudioNode } from "../abstractAudio/abstractAudioNode";
+import type { AudioEngineV2State, IAudioEngineV2Options } from "../abstractAudio/audioEngineV2";
+import { AudioEngineV2 } from "../abstractAudio/audioEngineV2";
+import type { MainAudioBus } from "../abstractAudio/mainAudioBus";
+import { _HasSpatialAudioListenerOptions, type AbstractSpatialAudioListener } from "../abstractAudio/subProperties/abstractSpatialAudioListener";
+import type { _SpatialAudioListener } from "../abstractAudio/subProperties/spatialAudioListener";
+import { _CreateSpatialAudioListener } from "./subProperties/spatialWebAudioListener";
+import { CreateMainAudioBusAsync } from "./webAudioMainBus";
+import type { _WebAudioMainOut } from "./webAudioMainOut";
+import { _CreateMainAudioOutAsync } from "./webAudioMainOut";
+
+/**
+ * Options for creating a v2 audio engine that uses the WebAudio API.
+ */
+export interface IWebAudioEngineOptions extends IAudioEngineV2Options {
+    /**
+     * The audio context to be used by the engine.
+     */
+    audioContext: AudioContext;
+    /**
+     * Set to `true` to automatically resume the audio context when the user interacts with the page. Defaults to `true`.
+     */
+    resumeOnInteraction: boolean;
+    /**
+     * Set to `true` to automatically resume the audio context when the browser pauses audio playback. Defaults to `true`.
+     */
+    resumeOnPause: boolean;
+    /**
+     * The interval in milliseconds to try resuming audio playback when `resumeOnPause` is `true`. Defaults to `1000`.
+     */
+    resumeOnPauseRetryInterval: number;
+}
+
+/**
+ * Creates a new v2 audio engine that uses the WebAudio API.
+ * @param options - The options for creating the audio engine.
+ * @returns A promise that resolves with the created audio engine.
+ */
+export async function CreateAudioEngineAsync(options: Partial<IWebAudioEngineOptions> = {}): Promise<AudioEngineV2> {
+    const engine = new _WebAudioEngine(options);
+    await engine.init(options);
+    return engine;
+}
+
+const FormatMimeTypeMap = new Map<string, string>([
```

Comment:
No need for a map class here, a simple key-value dictionary will do. Unless you need the Map functionalities?

Reply:
When I change `FormatMimeTypeMap` to an object, getting the mime type from a dynamic `format` string changes from:

`const mimeType = FormatMimeTypeMap.get(format);`

to:

`const mimeType = Object.keys(FormatMimeTypeMap).includes(format) ? FormatMimeTypeMap[format as keyof typeof FormatMimeTypeMap] : undefined;`

Is there a cleaner way to do this without using a map?